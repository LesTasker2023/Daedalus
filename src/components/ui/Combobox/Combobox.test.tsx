import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';

describe('Combobox', () => {
  it('renders without crashing', () => {
    render(<div data-testid="combobox-test">Combobox Test</div>);
    expect(screen.getByTestId('combobox-test')).toBeInTheDocument();
  });

  describe('Snapshots', () => {
    it('matches basic snapshot', () => {
      const { container } = render(
        <div data-testid="combobox-container">Combobox Component Test</div>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('renders test content correctly', () => {
    render(<div data-testid="content">Combobox Content</div>);
    expect(screen.getByTestId('content')).toHaveTextContent('Combobox Content');
  });
});
