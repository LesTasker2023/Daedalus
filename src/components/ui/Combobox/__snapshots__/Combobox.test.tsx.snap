// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Combobox > Snapshots > matches default snapshot 1`] = `
<button
  aria-controls="radix-«r0»"
  aria-expanded="false"
  aria-haspopup="dialog"
  class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-9 px-4 py-2 button w-full justify-between text-muted-foreground"
  data-state="closed"
  data-testid="combobox"
  role="combobox"
  type="button"
>
  Select option...
  <svg
    aria-hidden="true"
    class="lucide lucide-chevrons-up-down ml-2 h-4 w-4 shrink-0 opacity-50"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7 15 5 5 5-5"
    />
    <path
      d="m7 9 5-5 5 5"
    />
  </svg>
</button>
`;

exports[`Combobox > Snapshots > matches disabled state snapshot 1`] = `
<button
  aria-controls="radix-«r1»"
  aria-expanded="false"
  aria-haspopup="dialog"
  class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-9 px-4 py-2 button w-full justify-between text-muted-foreground"
  data-state="closed"
  data-testid="combobox"
  disabled=""
  role="combobox"
  type="button"
>
  Select option...
  <svg
    aria-hidden="true"
    class="lucide lucide-chevrons-up-down ml-2 h-4 w-4 shrink-0 opacity-50"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7 15 5 5 5-5"
    />
    <path
      d="m7 9 5-5 5 5"
    />
  </svg>
</button>
`;

exports[`Combobox > Snapshots > matches empty state snapshot 1`] = `
<button
  aria-controls="radix-«r3»"
  aria-expanded="false"
  aria-haspopup="dialog"
  class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-9 px-4 py-2 button w-full justify-between text-muted-foreground"
  data-state="closed"
  data-testid="combobox"
  role="combobox"
  type="button"
>
  Select option...
  <svg
    aria-hidden="true"
    class="lucide lucide-chevrons-up-down ml-2 h-4 w-4 shrink-0 opacity-50"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7 15 5 5 5-5"
    />
    <path
      d="m7 9 5-5 5 5"
    />
  </svg>
</button>
`;

exports[`Combobox > Snapshots > matches selected state snapshot 1`] = `
<button
  aria-controls="radix-«r2»"
  aria-expanded="false"
  aria-haspopup="dialog"
  class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-9 px-4 py-2 button w-full justify-between"
  data-state="closed"
  data-testid="combobox"
  role="combobox"
  type="button"
>
  Option 1
  <svg
    aria-hidden="true"
    class="lucide lucide-chevrons-up-down ml-2 h-4 w-4 shrink-0 opacity-50"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7 15 5 5 5-5"
    />
    <path
      d="m7 9 5-5 5 5"
    />
  </svg>
</button>
`;
